src/main/fc/core.c

```diff
@@ -205,6 +205,80 @@ bool canUseLaunchControl(void)
 }
 #endif
 
+static bool checkIfLowpassGetsDisabled(int nyquist, uint16_t filterCutoff)
+{
+    if (filterCutoff > nyquist) {
+        return true;
+    }
+    return false;
+}
+
+static bool checkIfGyroNotchGetsDisabled(int nyquist, uint16_t notchCenter, uint16_t notchCutoff)
+{
+    if (notchCenter > nyquist && calculateNyquistAdjustedNotchHz(notchCenter, notchCutoff) == 0) {
+        return true;
+    }
+    return false;
+}
+
+// Check to see if any gyro filtering is enabled
+static bool anyGyroFilteringEnabled(void)
+{
+    // Check gyro lowpass filters
+    if (gyroConfig()->dyn_lpf_gyro_min_hz > 0 || gyroConfig()->gyro_lowpass_hz > 0 || gyroConfig()->gyro_lowpass2_hz > 0) {
+        return true;
+    }
+
+    // Check gyro notch filters
+    if ((gyroConfig()->gyro_soft_notch_hz_1 > 0 && gyroConfig()->gyro_soft_notch_cutoff_1 > 0)
+        || (gyroConfig()->gyro_soft_notch_hz_2 > 0 && gyroConfig()->gyro_soft_notch_cutoff_2 > 0)) {
+        return true;
+    }
+
+    // Check dynamic filter
+    if (featureIsEnabled(FEATURE_DYNAMIC_FILTER)) {
+        return true;
+    }
+
+#ifdef USE_RPM_FILTER
+    // Check rpm filter
+    if (isRpmGyroFilterEnabled()) {
+        return true;
+    }
+#endif
+
+    // No gyro filtering!
+    return false;
+}
+
+// Check for invalid or disabled filtering and prevent
+// arming for unsafe settings.
+static void setFilteringArmingStatus(void)
+{
+    const int gyroFrequencyNyquist = 1000000 / 2 / gyro.targetLooptime;
+    const int pidFrequencyNyquist = gyroFrequencyNyquist / pidConfig()->pid_process_denom;
+    bool filterError = false;
+
+    // Check gyro filtering for cases of implicit disabling because of cutoffs being above the nyquist
+    uint16_t gyroLowpassHz = (gyroConfig()->dyn_lpf_gyro_min_hz > 0) ? gyroConfig()->dyn_lpf_gyro_min_hz : gyroConfig()->gyro_lowpass_hz;
+    filterError = filterError || checkIfLowpassGetsDisabled(gyroFrequencyNyquist, gyroLowpassHz);
```

Comment:
I still can't warm myself up to using the loop nyquist frequency as the limit at which we decide that 'this isn't enough filtering'. The loop frequency is chosen by the user based on considerations that got nothing to do with filtering. If the filters are above the nyquist frequency they get disabled because the math stops working **and because they are not needed**  as the nyquist theorem says that there already is a lowpass acting at the nyquist frequency.
So if we think that a lowpass that is at or above a certain frequency is 'not enough filtering' we should disable arming if the filters are above this (fixed) frequency, and not use the arbitrarily set nyquist frequency for this.

Reply:
The entire point of the "filter disabled" checks is to warn users that the filter settings they thought they configured are in fact not working due to another seemingly unrelated setting (loop time). There is no currently no indication that the filters have been implicitly disabled and that the user should reevaluate and adjust the cutoffs - that's what the arming disabled is providing. The filter would have been completely appropriate and working before the loop time change. It might also be nice for the configurator to enforce context-sensitive limits based on the loop times from a UX perspective, but this wouldn't help with CLI, CMS, Lua, etc. So catching it after the fact is the only way we have.

Regarding the second part of the checks for "not enough filtering" - I'm perfectly open to using this to enforce more reasonable "useful" cutoff limits rather than the "hard" mathematical limits imposed by nyquist. I'm just not sure how to define "useful".

Reply:
@etracer65:

> The entire point of the "filter disabled" checks is to warn users that the filter settings they thought they configured are in fact not working 

But the entire point of how the nyquist limit works is that **the frequencies that the filters would have caught up are filtered out by the nyquist limit** if the filter is entirely above the nyquist limit. So warning the users that 'the filters are not working' does not seem to be right if what they are getting is at least as much filtering if not more as if the filters were working and the nyquist limit above them.

> The filter would have been completely appropriate and working before the loop time change.

In this case the filtering will be appropriate if not better after the loop time change and disabling of filters caused by it, thanks to the nyquist limit.